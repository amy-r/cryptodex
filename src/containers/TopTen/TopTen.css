.TopTen {
  background: none;
  z-index: 100;
  position: relative;
  border: 3px solid white;
  border-radius: 10px;
  margin: auto;
  width: max-content;
  color: white;
}

.table-head {
  font-weight: 700;
}

td {
  margin: 10%;
}

tbody {
  margin: auto;
}

@media screen and (max-width: 967px) {
  .TopTen {
    width: min-content;
  }
}