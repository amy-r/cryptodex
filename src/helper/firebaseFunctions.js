import * as firebase from 'firebase';
import { config } from '../firebase/firebase';
export const firebaseApp = firebase.initializeApp(config);

export const writeCurrency = (user, curr) => {
  try {
    firebaseApp.database().ref(user).update({
      [curr.name]: curr.amount,
    });
  } catch (error) {
    throw new Error(`error adding currency : ${error}` )
  }
}

