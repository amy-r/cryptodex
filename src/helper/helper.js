export const createMarketCap = (dashboard) => {
  const newDash = dashboard.map( (currency) => {
    currency['marketCap']= parseFloat(currency.close * currency.availableSupply);
    return currency
  });
  return newDash
}

export const sortByCaps = (dashboard) => {
  return dashboard.sort(function(a,b) {
    return b.marketCap - a.marketCap
  })
}

export const trimData = (dashboard) => {
  return dashboard.map( (curr) => {
    return {
      'currency': curr.currency, 
      'marketCap': curr.marketCap
    }
  })
}

export const totalCap = (dashboard) => {
  const total = dashboard.reduce((acc, curr) => {
    return acc + curr.marketCap
  }, 0)
  return total
} 

export const createCapPercentage = (dashboard) => {
  const total = totalCap(dashboard)
  const finalCurrency = dashboard.map( (currency) => {
    const percentage = (currency.marketCap/total) * 100;
    const roundedPercentage = parseFloat(Math.max(percentage).toFixed(2))
    currency['percentage'] = roundedPercentage
    return currency
  })
  return finalCurrency
}