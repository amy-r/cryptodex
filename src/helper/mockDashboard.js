export const mockDashboard = [
  { 'currency' : "BTC",
    'price' : "7436.82313",
    'availableSupply' : "16957550.0",
    'marketCap' : 126110300068.1315 },
  { 'currency' : "ETH",
    'price' : "387.43253",
    'availableSupply' : "98680455.0",
    'marketCap' : 38207824855.0488 },
  { 'currency' : "XRP",
    'price' : "0.49032",
    'availableSupply' : "39094520623.0",
    'marketCap' : 19161006447.74476 },
]

export const mockDashboard2 = [
  { 'currency' : "BTC",
    'close' : "7436.82313",
    'availableSupply' : "16957550.0",
    'marketCap' : 126110300068.1315 },
  { 'currency' : "ETH",
    'close' : "387.43253",
    'availableSupply' : "98680455.0",
    'marketCap' : 38207824855.0488 },
  { 'currency' : "XRP",
    'close' : "0.49032",
    'availableSupply' : "39094520623.0",
    'marketCap' : 19161006447.74476 },
]