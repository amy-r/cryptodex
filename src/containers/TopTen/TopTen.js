import React from 'react';
import { connect } from 'react-redux';
import './TopTen.css';
import PropTypes from 'prop-types';

export const TopTen = (props) => {

  const tableData = props.dashboard.map( (curr) => {
    return (      
      <tr key= {curr.currency}>
        <td> {curr.currency} </td>
        <td> {curr.marketCap} </td>
        <td> {curr.percentage} </td>
      </tr>      
    );
  });

  return (
    <div className='TopTen'>
      <table>
        <tbody>
          <tr className='table-head'>
            <td> Currency </td>
            <td> Market Cap </td>
            <td> % </td>
          </tr>
          {tableData}
        </tbody>
      </table>
    </div>
  );
};

export const mapStateToProps = (state) => ({
  dashboard: state.dashboard
});

TopTen.propTypes = {
  dashboard: PropTypes.arrayOf(
    PropTypes.objectOf(PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number])))
};

export default connect(mapStateToProps, null)(TopTen);

