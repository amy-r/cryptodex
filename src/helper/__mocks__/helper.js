/* eslint-disable */

import {mockDashboard} from '../../helper/mockDashboard';

export const createMarketCap = jest.fn().mockImplementation( () => {
  return [
      { 'currency' : "BTC",
        'close' : "7436.82313",
        'availableSupply' : "16957550.0",
        'marketCap' : 126110300068.1315 }
    ]
})

export const sortByCaps = jest.fn().mockImplementation( () => {
  return mockDashboard
})

export const trimData = jest.fn().mockImplementation( () => {
  return mockDashboard
})


export const createCapPercentage = jest.fn().mockImplementation( () => {
  return mockDashboard
})

export const removeMismatches = jest.fn().mockImplementation( () => {
  return { BTC: 1, ETH: 1}
})

export const findMismatches = jest.fn().mockImplementation( () => {
  return { FAKECOIN: 1 }
})

export const calculateValue = jest.fn().mockImplementation( () => {
  return [7436.82, 387.43,]
})

export const calculatePercent = jest.fn().mockImplementation( () => {
  return [6.67,13.33,20,26.67,33.33]
})