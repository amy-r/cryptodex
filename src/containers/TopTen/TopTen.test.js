import React from 'react';
import { shallow } from 'enzyme';
import { TopTen, mapStateToProps } from './TopTen';
import { mockDashboard } from '../../helper/mockDashboard'; 

describe('TopTen', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<TopTen dashboard={mockDashboard}/>)
    expect(wrapper).toMatchSnapshot();
  })
})

describe('MSTP', () => {
  it('should define props for the container', () => {
    const mockStore = { dashboard: mockDashboard }
    const mapped = mapStateToProps(mockStore)
    expect(mapped.dashboard).toEqual(mockDashboard)
  })
})