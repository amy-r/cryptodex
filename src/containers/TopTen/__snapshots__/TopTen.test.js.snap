// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTen should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    dashboard={
        Array [
            Object {
              "availableSupply": "16957550.0",
              "currency": "BTC",
              "marketCap": 126110300068.1315,
              "price": "7436.82313",
            },
            Object {
              "availableSupply": "98680455.0",
              "currency": "ETH",
              "marketCap": 38207824855.0488,
              "price": "387.43253",
            },
            Object {
              "availableSupply": "39094520623.0",
              "currency": "XRP",
              "marketCap": 19161006447.74476,
              "price": "0.49032",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <table>
        <tbody>
                <tr>
                        <td>
                                 Currency 
                        </td>
                        <td>
                                 Market Cap 
                        </td>
                        <td>
                                 % 
                        </td>
                </tr>
                <tr>
                        <td>
                                 
                                BTC
                                 
                        </td>
                        <td>
                                 
                                126110300068.1315
                                 
                        </td>
                        <td>
                                 
                                 
                        </td>
                </tr>
                <tr>
                        <td>
                                 
                                ETH
                                 
                        </td>
                        <td>
                                 
                                38207824855.0488
                                 
                        </td>
                        <td>
                                 
                                 
                        </td>
                </tr>
                <tr>
                        <td>
                                 
                                XRP
                                 
                        </td>
                        <td>
                                 
                                19161006447.74476
                                 
                        </td>
                        <td>
                                 
                                 
                        </td>
                </tr>
        </tbody>
</table>,
      "className": "TopTen",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <tbody>
          <tr>
                    <td>
                               Currency 
                    </td>
                    <td>
                               Market Cap 
                    </td>
                    <td>
                               % 
                    </td>
          </tr>
          <tr>
                    <td>
                               
                              BTC
                               
                    </td>
                    <td>
                               
                              126110300068.1315
                               
                    </td>
                    <td>
                               
                               
                    </td>
          </tr>
          <tr>
                    <td>
                               
                              ETH
                               
                    </td>
                    <td>
                               
                              38207824855.0488
                               
                    </td>
                    <td>
                               
                               
                    </td>
          </tr>
          <tr>
                    <td>
                               
                              XRP
                               
                    </td>
                    <td>
                               
                              19161006447.74476
                               
                    </td>
                    <td>
                               
                               
                    </td>
          </tr>
</tbody>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <tr>
              <td>
                             Currency 
              </td>
              <td>
                             Market Cap 
              </td>
              <td>
                             % 
              </td>
</tr>,
            Array [
              <tr>
                <td>
                                 
                                BTC
                                 
                </td>
                <td>
                                 
                                126110300068.1315
                                 
                </td>
                <td>
                                 
                                 
                </td>
</tr>,
              <tr>
                <td>
                                 
                                ETH
                                 
                </td>
                <td>
                                 
                                38207824855.0488
                                 
                </td>
                <td>
                                 
                                 
                </td>
</tr>,
              <tr>
                <td>
                                 
                                XRP
                                 
                </td>
                <td>
                                 
                                19161006447.74476
                                 
                </td>
                <td>
                                 
                                 
                </td>
</tr>,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <td>
                   Currency 
</td>,
                <td>
                   Market Cap 
</td>,
                <td>
                   % 
</td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Currency ",
                },
                "ref": null,
                "rendered": " Currency ",
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Market Cap ",
                },
                "ref": null,
                "rendered": " Market Cap ",
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " % ",
                },
                "ref": null,
                "rendered": " % ",
                "type": "td",
              },
            ],
            "type": "tr",
          },
          Object {
            "instance": null,
            "key": "BTC",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <td>
                   
                  BTC
                   
</td>,
                <td>
                   
                  126110300068.1315
                   
</td>,
                <td>
                   
                   
</td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "BTC",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "BTC",
                  " ",
                ],
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    126110300068.1315,
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  126110300068.1315,
                  " ",
                ],
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    undefined,
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  undefined,
                  " ",
                ],
                "type": "td",
              },
            ],
            "type": "tr",
          },
          Object {
            "instance": null,
            "key": "ETH",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <td>
                   
                  ETH
                   
</td>,
                <td>
                   
                  38207824855.0488
                   
</td>,
                <td>
                   
                   
</td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "ETH",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "ETH",
                  " ",
                ],
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    38207824855.0488,
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  38207824855.0488,
                  " ",
                ],
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    undefined,
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  undefined,
                  " ",
                ],
                "type": "td",
              },
            ],
            "type": "tr",
          },
          Object {
            "instance": null,
            "key": "XRP",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <td>
                   
                  XRP
                   
</td>,
                <td>
                   
                  19161006447.74476
                   
</td>,
                <td>
                   
                   
</td>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "XRP",
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "XRP",
                  " ",
                ],
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    19161006447.74476,
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  19161006447.74476,
                  " ",
                ],
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    undefined,
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  undefined,
                  " ",
                ],
                "type": "td",
              },
            ],
            "type": "tr",
          },
        ],
        "type": "tbody",
      },
      "type": "table",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <table>
          <tbody>
                    <tr>
                              <td>
                                         Currency 
                              </td>
                              <td>
                                         Market Cap 
                              </td>
                              <td>
                                         % 
                              </td>
                    </tr>
                    <tr>
                              <td>
                                         
                                        BTC
                                         
                              </td>
                              <td>
                                         
                                        126110300068.1315
                                         
                              </td>
                              <td>
                                         
                                         
                              </td>
                    </tr>
                    <tr>
                              <td>
                                         
                                        ETH
                                         
                              </td>
                              <td>
                                         
                                        38207824855.0488
                                         
                              </td>
                              <td>
                                         
                                         
                              </td>
                    </tr>
                    <tr>
                              <td>
                                         
                                        XRP
                                         
                              </td>
                              <td>
                                         
                                        19161006447.74476
                                         
                              </td>
                              <td>
                                         
                                         
                              </td>
                    </tr>
          </tbody>
</table>,
        "className": "TopTen",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tbody>
            <tr>
                        <td>
                                     Currency 
                        </td>
                        <td>
                                     Market Cap 
                        </td>
                        <td>
                                     % 
                        </td>
            </tr>
            <tr>
                        <td>
                                     
                                    BTC
                                     
                        </td>
                        <td>
                                     
                                    126110300068.1315
                                     
                        </td>
                        <td>
                                     
                                     
                        </td>
            </tr>
            <tr>
                        <td>
                                     
                                    ETH
                                     
                        </td>
                        <td>
                                     
                                    38207824855.0488
                                     
                        </td>
                        <td>
                                     
                                     
                        </td>
            </tr>
            <tr>
                        <td>
                                     
                                    XRP
                                     
                        </td>
                        <td>
                                     
                                    19161006447.74476
                                     
                        </td>
                        <td>
                                     
                                     
                        </td>
            </tr>
</tbody>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <tr>
                <td>
                                 Currency 
                </td>
                <td>
                                 Market Cap 
                </td>
                <td>
                                 % 
                </td>
</tr>,
              Array [
                <tr>
                  <td>
                                     
                                    BTC
                                     
                  </td>
                  <td>
                                     
                                    126110300068.1315
                                     
                  </td>
                  <td>
                                     
                                     
                  </td>
</tr>,
                <tr>
                  <td>
                                     
                                    ETH
                                     
                  </td>
                  <td>
                                     
                                    38207824855.0488
                                     
                  </td>
                  <td>
                                     
                                     
                  </td>
</tr>,
                <tr>
                  <td>
                                     
                                    XRP
                                     
                  </td>
                  <td>
                                     
                                    19161006447.74476
                                     
                  </td>
                  <td>
                                     
                                     
                  </td>
</tr>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                     Currency 
</td>,
                  <td>
                     Market Cap 
</td>,
                  <td>
                     % 
</td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Currency ",
                  },
                  "ref": null,
                  "rendered": " Currency ",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Market Cap ",
                  },
                  "ref": null,
                  "rendered": " Market Cap ",
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " % ",
                  },
                  "ref": null,
                  "rendered": " % ",
                  "type": "td",
                },
              ],
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": "BTC",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                     
                    BTC
                     
</td>,
                  <td>
                     
                    126110300068.1315
                     
</td>,
                  <td>
                     
                     
</td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "BTC",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "BTC",
                    " ",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      126110300068.1315,
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    126110300068.1315,
                    " ",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      undefined,
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    undefined,
                    " ",
                  ],
                  "type": "td",
                },
              ],
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": "ETH",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                     
                    ETH
                     
</td>,
                  <td>
                     
                    38207824855.0488
                     
</td>,
                  <td>
                     
                     
</td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "ETH",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "ETH",
                    " ",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      38207824855.0488,
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    38207824855.0488,
                    " ",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      undefined,
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    undefined,
                    " ",
                  ],
                  "type": "td",
                },
              ],
              "type": "tr",
            },
            Object {
              "instance": null,
              "key": "XRP",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <td>
                     
                    XRP
                     
</td>,
                  <td>
                     
                    19161006447.74476
                     
</td>,
                  <td>
                     
                     
</td>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "XRP",
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "XRP",
                    " ",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      19161006447.74476,
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    19161006447.74476,
                    " ",
                  ],
                  "type": "td",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      undefined,
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    undefined,
                    " ",
                  ],
                  "type": "td",
                },
              ],
              "type": "tr",
            },
          ],
          "type": "tbody",
        },
        "type": "table",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
